### PR TITLE
Updated/added few tests related to host_unification

### PR DESCRIPTION
**Pull request by sghai**
These tests are part of host_unification feature so I renamed the test module test_host_system_unification.py` to  `test_hostunification.py` and added/updated tests.

Since this feature includes stuff around migration and unification of hosts and content-hosts, so chances are we need to move tests from content-hosts to hosts module with less addition of new tests. But that's applicable to tests via api and cli.

### CODE REVIEW

**Comment by oshtaier** [comment]
Please update `tests.foreman.ui.rst` file

File: tests/foreman/ui/test_hostunification.py [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+        @steps:
+        1.  Register a host via rhsm using credentials
+        2.  View host under content hosts
+        3.  View host under 'All Hosts'
+
+        @assert: Hosts registered via rhsm appears under 'All hosts' as well
+        as under content-hosts.
+
+        @status: Manual
+
```
Comment (elyezer) [inline tests/foreman/ui/test_hostunification.py]: Please drop all blank lines before closing `"""`. They are not necessary.

**Comment by elyezer** [comment]
ACK, pending @oshtaier comment and my non-blocking comment.

**Comment by sghai** [comment]
@elyezer @oshtaier : comments have been addressed. thanks

**Comment by oshtaier** [comment]
ACK

**Pull request merged**